Feature.friendlier parse errors

jneen opened the pull request:
These are some surface-level changes to parse errors, as well as a change to the tokenizer to allow whitespace between `1 .. 100` ranges, since I saw config getting tripped up on this.

YenTheFirst commented:
I think another part of the problem with dicts is that it's parsing commas differently than how lists or arguments parse commas.

https://github.com/opencounter/dentaku/blob/type.checker/lib/dentaku/parser.rb#L237
vs
https://github.com/opencounter/dentaku/blob/type.checker/lib/dentaku/parser.rb#L274
vs
https://github.com/opencounter/dentaku/blob/type.checker/lib/dentaku/parser.rb#L254

I'm really not sure why lists push an arity, but dictionaries don't.

davidhampgonsalves commented:
Yea this was changed pretty heavily upstream and I think it works better
there.

I think you mentioned that the type checker work made pulling those changes
in difficult though right?

On Thu, Jul 20, 2017 at 5:38 PM, T S <notifications@github.com> wrote:

> I think another part of the problem with dicts is that it's parsing commas
> differently than how lists or arguments parse commas.
>
> https://github.com/opencounter/dentaku/blob/type.
> checker/lib/dentaku/parser.rb#L237
> vs
> https://github.com/opencounter/dentaku/blob/type.
> checker/lib/dentaku/parser.rb#L274
> vs
> https://github.com/opencounter/dentaku/blob/type.
> checker/lib/dentaku/parser.rb#L254
>
> I'm really not sure why lists push an arity, but dictionaries don't.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/opencounter/dentaku/pull/6#issuecomment-316822934>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAK6zZI3bgUH8osyuH3dk0j_nHLzgjmkks5sP7q3gaJpZM4OefKz>
> .
>

YenTheFirst commented:
I haven't looked at merging in upstream changes, at all.

joshuabates commented:
I don't think we'll be able to merge in anything upstream at this point. We rewrote the tokenizer from scratch, and did a major refactoring of the parser.

Commit:
allow spaces surrounding numeric ranges

TODO: take this out of the tokenizer and make this actually parsed

Commit:
better error message for unknown token

previously we dumped half the expression into the message. we already
store the location so we should be able to highlight it in the editor
itself.

Commit:
allow multilingual characters in identifiers

using ruby's [[:alnum:]]
